src/System.Runtime.Analyzers/Core/AttributeStringLiteralsShouldParseCorrectly.cs

```diff
@@ -25,24 +29,232 @@ public abstract class AttributeStringLiteralsShouldParseCorrectlyAnalyzer : Diag
                                                                              customTags: WellKnownDiagnosticTags.Telemetry);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(DefaultRule, EmptyRule);
 
+        private static List<ValueValidator> s_tokensToValueValidator =
+            new List<ValueValidator>(
+                new[] { new ValueValidator(new[] { "guid" }, "Guid", GuidValueValidator),
+                        new ValueValidator(new[] { "url", "uri", "urn" }, "Uri", UrlValueValidator, "UriTemplate")});
+
+        private static bool GuidValueValidator(string value)
+        {
+            try
+            {
+                new Guid(value);
+                return true;
+            }
+            catch (OverflowException)
+            {
+            }
+            catch (FormatException)
+            {
+            }
+
+            return false;
+        }
+
+        private static bool UrlValueValidator(string value)
+        {
+            return Uri.IsWellFormedUriString(value, System.UriKind.RelativeOrAbsolute);
+        }
+
         public override void Initialize(AnalysisContext analysisContext)
         {
+            analysisContext.RegisterSymbolAction(saContext =>
+            {
+                var symbol = saContext.Symbol;
+                AnalyzeSymbol(saContext.ReportDiagnostic, symbol);
+                switch(symbol.Kind)
+                {
+                    case SymbolKind.NamedType:
+                        {
+                            var namedType = symbol as INamedTypeSymbol;
+
+                            AnalyzeSymbols(saContext.ReportDiagnostic, namedType.TypeParameters);
+
+                            if (namedType.TypeKind == TypeKind.Delegate && namedType.DelegateInvokeMethod != null)
+                            {
+                                AnalyzeSymbols(saContext.ReportDiagnostic, namedType.DelegateInvokeMethod.Parameters);
+                            }
+
+                            return;
+                        }
+
+                    default:
+                        {
+                            var methodSymbol = symbol as IMethodSymbol;
+                            if (methodSymbol != null && !methodSymbol.IsAccessorMethod())
+                            {
+                                AnalyzeSymbols(saContext.ReportDiagnostic, methodSymbol.Parameters);
+                                AnalyzeSymbols(saContext.ReportDiagnostic, methodSymbol.TypeParameters);
+                            }
+
+                            var propertySymbol = symbol as IPropertySymbol;
+                            if (propertySymbol != null)
+                            {
+                                AnalyzeSymbols(saContext.ReportDiagnostic, propertySymbol.Parameters);
+                            }
```

Comment:
They can be. As I was addressing different case, the code diverged. I will modify the code.